File: src/examples/javascript/hello-world.js
/**
 * Hello World - JavaScript
 *
 * Classic Hello World with variable declarations and string operations.
 * Perfect for beginners to see their first trace.
 */
export default {
  id: 'hello-world-js',
  language: 'javascript',
  title: 'Hello, World!',
  description: 'The classic beginner example — trace a Hello World program',
  difficulty: 'beginner',
  category: 'basics',
  tags: ['beginner', 'basics', 'hello-world'],
  code: `// Classic Hello World
const greeting = "Hello";
const name = "World";
const message = greeting + ", " + name + "!";

console.log(message);

// String methods
const upper = message.toUpperCase();
const length = message.length;

console.log("Uppercase:", upper);
console.log("Length:", length, "characters");

// Template literals
const year = 2026;
console.log(\`Welcome to NitinTrace in \${year}!\`);`,
};

File: src/examples/java/HelloWorld.java
export default {
  id: 'hello-world-java',
  language: 'java',
  title: 'Hello World (Java)',
  description: 'Simple Java class with methods and output',
  difficulty: 'beginner',
  category: 'basics',
  tags: ['basics', 'oop', 'class'],
  code: `public class HelloWorld {
  private String message;

  public HelloWorld(String message) {
    this.message = message;
  }

  public void greet() {
    System.out.println(message);
    System.out.println("Current time: " + System.currentTimeMillis());
  }

  public static void main(String[] args) {
    HelloWorld hw = new HelloWorld("Hello, World!");
    hw.greet();
  }
}`
};
